stages/org.osbuild.ostree.coreos.aleph

```diff
@@ -0,0 +1,161 @@
+
+    aleph_version_data = {
+        "osbuild-version": osbuild.__version__,
+        "version": aleph_version,
+        "ref": aleph_ref,
+        "ostree-commit": aleph_ostree_commit
+    }
+
+    write_to_file(tree, ".aleph-version.json", aleph_version_data, coreos_compat)
+
+
+def aleph_container(tree, imgref, coreos_compat):
+    # extract the image name from the imgref
+    imgref_list = imgref.split(':')
+    if imgref_list[0] in ["ostree-remote-registry", "ostree-remote-image"]:
+        img_name = ':'.join(imgref_list[2:])
+    elif imgref_list[0] in ["ostree-image-signed", "ostree-unverified-registry"]:
+        img_name = ':'.join(imgref_list[1:])
+
+    img_name = img_name.lstrip('docker://')
+
+    extra_args = []
+    extra_args.append(f"--repo={tree}/ostree/repo")
+    extra_args.append(f"registry:{img_name}")
+
+    container_data_json = ostree.cli("container", "image", "metadata", *extra_args).stdout.rstrip()
+    container_data = json.loads(container_data_json)
+
+    extra_args.append("--config")
+    container_data_config_json = ostree.cli("container", "image", "metadata", *extra_args).stdout.rstrip()
+    container_data_config = json.loads(container_data_config_json)
+
+    aleph_digest = container_data['config']['digest']
+    aleph_ref = f"docker://{imgref}"
+    aleph_version = container_data_config['config']['Labels']['org.opencontainers.image.version']
+    aleph_container_image = container_data_config['config']['Labels']
+
+    aleph_version_data = {
+        "osbuild-version": osbuild.__version__,
+        "ref": aleph_ref,
+        "build": aleph_version,
+        "version": aleph_version,
+        "container-image": {
+            "image-name": img_name,
+            "image-digest": aleph_digest,
+            "image-labels": aleph_container_image
+        }
+    }
+    
+    # the 'ostree.commit' label will be optional in the future so
+    # prevent hard failing if key is not found
+    aleph_ostree_commit = container_data_config['config']['Labels'].get('ostree.commit')
+    if aleph_ostree_commit is not None:
+        aleph_version_data["ostree-commit"] = aleph_ostree_commit
+
+    write_to_file(tree, ".aleph-version.json", aleph_version_data, coreos_compat)
+
+
+def main(tree, options):
+    coreos_compat = options.get("coreos_compat", False)
+    dep = options.get("deployment", None)
+    osname = ""
+    ref = ""
+    serial = None
+    origin = ""
+
+    # if deployment is specified, use it to find the origin file.
+    # otherwise, autodetect the only deployment found in the tree.
+    if dep is not None:
+        osname = dep.get("osname", "")
+        ref = dep.get("ref", "")
+        serial = dep.get("serial", 0)
+    
+    origin = ostree.deployment_path(tree, osname, ref, serial) + ".origin"
+    
+    # example container case: docker://quay.io/fedora/fedora-coreos:stable
+    # example ostree commit case: fedora:fedora/x86_64/coreos/stable
```

Comment:
just a little more here would be better:

```suggestion
    # example container case: container-image-reference=ostree-remote-image:fedora:docker://quay.io/fedora/fedora-coreos:stable
    # example ostree commit case: baserefspec=fedora:fedora/x86_64/coreos/stable
```

looking at the file it also appears that we are relying on the `baserefspec=` or `container-image-reference=` being the first line in the file with `=` in it. Is that right? Maybe we should parse the file line by line until we get to one that starts with one of our triggering conditions?

That should probably also be what we use to dictate `aleph_container` or `aleph_commit` below too.

Reply:
Sounds good. Just FYI, I had a look at the origin file for an ostree commit deployed systems and I'm pretty sure it's `refspec` not `baserefspec` (which is why I changed it).

Reply:
ahh. that's interesting. In my case I have layered packages so that's probably why it's different:

```
[core@apu2 ~]$ cat /ostree/deploy/fedora-coreos/deploy/164a9a34590640f49fd9d8258f236d0b5097872464f11fb968dc40e04e1a51d4.0.origin 
[origin]
baserefspec=fedora:fedora/x86_64/coreos/testing
override-commit=a47a9bb382283dc8adcd65a0fa54a570d2eb684d988cbd6bd2988d40d7fb9565

[packages]
requested=NetworkManager-wifi;bridge-utils;dmidecode;firewalld;flashrom;iwd;libimobiledevice;libimobiledevice-utils;lshw;pciutils;speedtest-cli;systemd-oomd-defaults;tailscale;tmux;usbmuxd;
```

We probably don't need to handle that case, though, because here we are creating images and this is something that would happen day2 by the user.